packages/babel-core/src/config/plugin.js

```diff
@@ -31,7 +34,8 @@ export default class Plugin {
     this.pre = plugin.pre;
     this.visitor = plugin.visitor;
     this.capabilities = plugin.capabilities;
-    this.dependencies = plugin.dependencies;
+    this.after = plugin.after;
+    this.before = plugin.before;
```

Comment:
based on my [prev comment](https://github.com/babel/babel/pull/5735/commits/3e27d174ee85f24919aed0d78227ba2015dc6117#r125368481), might wanna do a `this.x = plugin.x || [];` to account for cases where its not present?